DetectorDescription/Core/src/DDCompactView.cc

```diff
@@ -100,6 +110,20 @@ void DDCompactView::lockdown() {
   DDSpecifics::StoreT::instance().swap(specStore_);
   DDRotation::StoreT::instance().swap(rotStore_);
 
+  //Deal with DDVectors
+  DDVector::iterator<DDVector> vit;
+  DDVector::iterator<DDVector> ved(DDVector::end());
+
+  for (; vit != ved; ++vit) {
+    if (vit->isDefined().second) {
+      DDName vname(vit->name());
+      vectors_.emplace(vname.name(), vit->values());
+    }
+  }
+  // NOTE: would be good to clear DDVector's container at this time
+  // to avoid mixing with other DDCompactViews. Unclear if that is
+  // safe to do.
+
```

Comment:
@Dr15Jones - IMHO, there is no harm in trying to zap it. That is the only way to see if there are any `DDName`d objects that could still refer to the `DDVector`s. There shouldn't be. Oh, wait! The MagneticField built from the XML might need this access :-(
That is also the reason for not locking the Stores...
One drawback is that there will be a map of the same `Vector`s per `DDCompactView`.

Reply:
I had realised we'd get multiple copies. Given how little used the `DDVectorGetter` actually was (and I couldn't find any direct uses of `DDVector` except where they were being created) I figured the map probably wasn't going to be too full.